File: app/src/main/java/com/ytempest/wanandroid/interactor/configs/base/PreferencesName.java
package com.ytempest.wanandroid.interactor.configs.base;

import androidx.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @author heqidu
 * @since 2020/8/15
 */
@StringDef({
        PreferencesName.USER,
})
@Retention(RetentionPolicy.SOURCE)
public @interface PreferencesName {
    String USER = "config_user";
}

File: app/src/main/java/com/ytempest/wanandroid/interactor/configs/base/PreferencesName.kt
package com.ytempest.wanandroid.interactor.configs.base

import androidx.annotation.StringDef

/**
 * @author heqidu
 * @since 21-2-8
 */
@StringDef(
        PreferencesName.USER,
)
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class PreferencesName {
    companion object {
        const val USER = "config_user"
    }
}